src/components/Popover/src/Popover.vue

```diff
@@ -0,0 +1,216 @@
+	</div>
+</template>
+
+<script>
+import { throwError } from '@square/maker/utils/debug';
+import { MTransitionFadeIn } from '@square/maker/components/TransitionFadeIn';
+import { Portal } from '@linusborg/vue-simple-portal';
+import { PopoverConfigKey, PopoverAPIKey } from './keys';
+import PopoverInstance from './PopoverInstance.vue';
+import { generateRandomId } from './utils';
+
+const MAX_TETHER_VNODE = 1;
+const SKIDDING_OFFSET = 0;
+const DISTANCE_OFFSET = 8;
+
+const DEFAULT_MODIFIERS = [
+	{
+		name: 'offset',
+		options: { offset: [SKIDDING_OFFSET, DISTANCE_OFFSET] },
+	},
+	{
+		name: 'preventOverflow',
+		options: {
+			padding: 0,
+			altBoundary: true,
+		},
+	},
+	{
+		name: 'minWidth',
+		enabled: true,
+		phase: 'beforeWrite',
+		requires: ['computeStyles'],
+		fn({ state: { styles, rects } }) {
+			styles.popper.minWidth = `${rects.reference.offsetWidth}px`;
+		},
+		effect({ state: { elements } }) {
+			elements.popper.style.minWidth = `${elements.reference.offsetWidth}px`;
+		},
+	},
+];
+
+export default {
+	name: 'Popover',
+
+	components: {
+		Portal,
+		PopoverInstance,
+		MTransitionFadeIn,
+	},
+
+	inject: {
+		popoverApi: {
+			from: PopoverAPIKey,
+			default: () => throwError('No popover layer detected', 'Popover'),
+		},
+
+		popoverConfig: {
+			from: PopoverConfigKey,
+			default: {
+				config: undefined,
+			},
+		},
+	},
+
+	props: {
+		/**
+		 * Starting position of the popover. Not guaranteed if too close to overflow.
+		 */
+		placement: {
+			type: String,
+			default: 'bottom-start',
+			validator: (placement) => [
+				'auto', 'auto-start', 'auto-end',
+				'top', 'top-start', 'top-end',
+				'right', 'right-start', 'right-end',
+				'bottom', 'bottom-start', 'bottom-end',
+				'left', 'left-start', 'left-end',
+			].includes(placement),
+		},
+
+		/**
+		 * Popper.js modifier list. Overrides default configuration if present.
+		 */
+		modifiers: {
+			type: Array,
+			default: () => DEFAULT_MODIFIERS,
+		},
+	},
+
+	data() {
+		const vm = this;
+
+		return {
+			/**
+			 * This is used to track which popover is open. When opening, this
+			 * will be pushed into the popoverApi state as the opened popover.
+			 */
+			id: generateRandomId(),
+			currentPopper: undefined,
+			popperToDestroy: undefined,
+			actionAPI: {
+				open(...ignoreElements) {
+					if (vm.isOpen) {
+						return;
+					}
+
+					vm.popoverApi.setPopover({
+						tetherEl: vm.tetherEl,
+						ignoreEls: ignoreElements,
+						id: vm.id,
+					});
+				},
+
+				close() {
+					this.popperToDestroy = this.currentPopper;
+					this.currentPopper = undefined;
+					vm.popoverApi.closePopover();
+				},
+
+				toggle(...ignoreElements) {
+					if (vm.isOpen) {
+						vm.actionAPI.close();
+					} else {
+						vm.actionAPI.open(...ignoreElements);
+					}
+				},
+			},
+		};
+	},
+
+	computed: {
+		tetherEl() {
+			if (this.$el.children.length !== MAX_TETHER_VNODE) {
+				return undefined;
+			}
+
+			return this.$el.children[0];
```

Comment:
You should be able to replace this with subslot. It's a utility to filter slots, but declaratively in the template.

Reply:
Interesting, I didn't know this was a thing!